File: frontend/src/stores/system.ts
import { getSystemStatus } from "@/api/api";

export const useSystemStore = defineStore("global", () => {
  const isSystemInit = ref(true);
  const version = ref("0.0.0");
  const isLogin = ref(false);

  async function loadSystemData() {
    const system = await getSystemStatus();
    version.value = system.version;
    isLogin.value = system.is_login;
    isSystemInit.value = system.is_system_init;
  }

  return {
    isSystemInit,
    version,
    isLogin,
    loadSystemData,
  };
});

File: internal/app/handler/system.go
package handler

import (
	"github.com/gin-gonic/gin"
	"github.com/simonkimi/minebangumi/api"
	"github.com/simonkimi/minebangumi/internal/app/config"
)

func (w *WebApi) systemStatus(c *gin.Context) {
	db := w.mgr.GetDatabase()
	conf := w.mgr.GetConfig()

	isLogin := IsLogin(c)
	username := ""
	if isLogin {
		username = conf.GetString(config.UserUsername)
	}

	api.OkResponse(c, &api.SystemInfo{
		Version:                api.Version,
		AppDatabaseVersion:     db.GetAppSchemaVersion(),
		CurrentDatabaseVersion: db.GetSchemaVersion(),
		IsSystemInit:           conf.GetBool(config.SystemInit),
		IsLogin:                isLogin,
		Username:               username,
	})
}
